### PR TITLE
Add reporter option to allow gauges for summed metrics

### PR DESCRIPTION
**Pull request by c-brenn**
👋 Hopefully the commit message below gives a decent explanation behind the motivations for making a change like this.

I've opened this PR to get your feedback and see if it's something that might be considered - feel free to close this out if not  😄 
Likewise if there's an alternative idea as to how to implement this I'd be more than happy to rework this PR.

Thanks for your work on this package, using it has been a breeze!

## Commit Message

Some summed metrics can decrease as well as increase, in which case a
Prometheus counter isn't suitable as they are monotonic. A Prometheus
gauge fits nicely as it allows for arbitrary increases and decreases
value.

At the moment, it's only possible to get a gauge from `last_value`,
meaning if you want to report a sum goes up and down, you currently need
to sum the value yourself and make it available to `last_value` via
something like `telemetry_metrics_poller`.

This commit allows users to pass the `:prometheus_type` reporter option
when declaring a `sum`, which determines whether the metric exported to
prometheus is a counter or a gauge.

### CODE REVIEW

File: lib/core.ex [hunk 1]
```diff
@@ -32,7 +32,7 @@ defmodule TelemetryMetricsPrometheus.Core do
     * Counter - Counter
     * Distribution - Histogram
     * LastValue - Gauge
-    * Sum - Counter
+    * Sum - Counter/Gauge
```
Comment (c-brenn) [inline lib/core.ex:35]: I wasn't sure how best to document a new reporter option, some guidance here would be appreciated to make sure this is discoverable for folks 🙇

Comment (bryannaegele) [reply lib/core.ex:35]: Maybe you could just add an example metric definition using the option in the example above. Looks great otherwise!

Comment (c-brenn) [reply lib/core.ex:36]: Thanks! I've added an example above and also a section outlining the currently supported `:reporter_options` further down. Let me know what you think 😄

**Comment by c-brenn** [comment]
@bryannaegele I updated the documentation a little while ago so folks can find the new option 😄  Let me know if there are any further improvements we'd like to get this change in a fit state to merge.

File: lib/core/exporter.ex [hunk 2]
```diff
@@ -24,7 +24,15 @@ defmodule TelemetryMetricsPrometheus.Core.Exporter do
   end
 
   def format(%Sum{} = metric, time_series) do
-    format_standard({metric, time_series}, "counter")
+    type =
+      metric.reporter_options
+      |> Keyword.get(
+        :preometheus_type,
```
Comment (bryannaegele) [inline lib/core/exporter.ex:30]:
```suggestion
        :prometheus_type,
```

File: test/exporter_test.exs [hunk 3]
```diff
@@ -151,6 +151,31 @@ defmodule TelemetryMetricsPrometheus.Core.ExporterTest do
...
+      # TYPE cache_key_invalidations_total gauge
+      cache_key_invalidations_total{name="users"} 1027
+      cache_key_invalidations_total{name="short_urls"} 3\
+      """
+
+      metric =
+        Metrics.sum("cache.key.invalidations.total",
+          tags: ["name"],
+          description: "The total number of cache key invalidations.",
+          reporter_options: [preometheus_type: :gauge]
```
Comment (bryannaegele) [inline test/exporter_test.exs:166]:
```suggestion
          reporter_options: [prometheus_type: :gauge]
```

File: test/registry_test.exs [hunk 4]
```diff
@@ -100,6 +100,25 @@ defmodule TelemetryMetricsPrometheus.Core.RegistryTest do
     end
   end
 
+  test "validates for preometheus_type" do
```
Comment (bryannaegele) [inline test/registry_test.exs:103]:
```suggestion
  test "validates for prometheus_type" do
```

File: test/registry_test.exs [hunk 5]
```diff
@@ -100,6 +100,25 @@ defmodule TelemetryMetricsPrometheus.Core.RegistryTest do
...
+    assert_raise ArgumentError, fn ->
+      Metrics.sum("some.plug.call.total", reporter_options: [prometheus_type: :invalid])
+      |> Registry.validate_prometheus_type!()
+    end
+
+    assert Metrics.sum("som.plug.call.total", reporter_options: [preometheus_type: :counter])
```
Comment (bryannaegele) [inline test/registry_test.exs:109]:
```suggestion
    assert Metrics.sum("some.plug.call.total", reporter_options: [prometheus_type: :counter])
```

File: test/registry_test.exs [hunk 6]
```diff
@@ -100,6 +100,25 @@ defmodule TelemetryMetricsPrometheus.Core.RegistryTest do
...
+           |> Registry.validate_prometheus_type!() == :ok
+
+    assert Metrics.sum("som.plug.call.total", reporter_options: [preometheus_type: :gauge])
```
Comment (bryannaegele) [inline test/registry_test.exs:112]:
```suggestion
    assert Metrics.sum("some.plug.call.total", reporter_options: [prometheus_type: :gauge])
```

File: test/registry_test.exs [hunk 7]
```diff
@@ -100,6 +100,25 @@ defmodule TelemetryMetricsPrometheus.Core.RegistryTest do
...
+           |> Registry.validate_prometheus_type!() == :ok
+
+    assert Metrics.sum("som.plug.call.total", reporter_options: [])
```
Comment (bryannaegele) [inline test/registry_test.exs:115]:
```suggestion
    assert Metrics.sum("some.plug.call.total", reporter_options: [])
```

File: test/registry_test.exs [hunk 8]
```diff
@@ -100,6 +100,25 @@ defmodule TelemetryMetricsPrometheus.Core.RegistryTest do
...
+           |> Registry.validate_prometheus_type!() == :ok
+
+    assert Metrics.sum("som.plug.call.total")
```
Comment (bryannaegele) [inline test/registry_test.exs:118]:
```suggestion
    assert Metrics.sum("some.plug.call.total")
```

**Review by bryannaegele — COMMENTED** [review]
Sorry for the delay. Just a few typos and I think we're good :)

**Comment by c-brenn** [comment]
Yikes, I need to be a bit more careful with my editor's tab-complete 😅 

Thanks for the review!

**Review by bryannaegele — APPROVED** [review]

**Pull request merged**